lib/pronto/formatter/formatter.rb

```diff
@@ -1,30 +1,32 @@
 module Pronto
   module Formatter
-    def self.get(names)
-      names ||= 'text'
-      Array(names).map { |name| FORMATTERS[name.to_s] || TextFormatter }
-        .uniq.map(&:new)
-    end
+    class << self
+      def add(label, formatter_klass = nil)
+        unless formatter_klass.method_defined?(:format)
+          raise NoMethodError, "format method is not declared in the #{label} class."
+        end
 
-    def self.names
-      FORMATTERS.keys
-    end
+        base = Pronto::Formatter::Base
+        raise "#{label.inspect} is not a #{base}" unless formatter_klass.ancestors.include?(base)
+
+        raise ArgumentError, "formatter #{label.inspect} has already been added." if formatters.key?(label)
+
```

Comment:
**concern**: I'm not so sure about this part, what happens if the code gets reloaded or required in some way which causes the same formatter to be added twice? Normally, this shouldn't be a concern but I'm a bit wary about guessing about the way people use Pronto.

Reply:
Yeah, makes sense.
I haven't thought about reloading the formatter.